src/core/messaging/ACSClient.ts

```diff
@@ -219,7 +219,7 @@ export class ACSConversation {
                 }
 
                 // Filter out duplicate messages
-                if (postedMessageIds.has(id)) {
+                if (postedMessageIds.has(id) && postedMessageIds.has(version)) {

```

Comment:
Where is `version` being added to `postedMessageIds`? And shouldn't we use `id + version` to identify a unique record inside `postedMessageIds`?

Reply:
After discussion with @xTEddie , updated logic to rely on `editedOn` property instead of `version`.